### PR TITLE
Replace Rectangle::expand with Polygon's

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* Removed Polygon::expand(Vector)
* Removed Rectangle::expand(Vector)
* Added Rectangle::expand(double)

### Testing Done
Moved tests from polygon to rectangle

### Resolved Issues
N/A

### Length Justification
N/A

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/geom/polygon.cpp [hunk 1]
```diff
@@ -49,76 +49,6 @@ Point Polygon::centroid() const
...
-    //          expansion vector
-    //
-    // Expanded Polygon (A and D are shifted down):
-    //
-    //            B---------C
-    //            |         |
-    //            |         |
-    //            |         |
-    //            A'--------D'
-    //
```
Comment (nimazareian) [inline src/software/geom/polygon.cpp:74]: Why do we not want to keep this functionality? It seems to be giving more control over the expansion compared to `Polygon::expand(double)`.

File: src/software/ai/hl/stp/tactic/goalie/simulated_goalie_tactic_test.cpp [hunk 2]
```diff
@@ -144,11 +144,8 @@ TEST_F(SimulatedGoalieTacticTest, test_slow_ball_at_sharp_angle_to_friendly_goal
     std::vector<ValidationFunction> terminating_validation_functions = {
         [this](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
             checkGoalieSuccess(5, world_ptr, yield);
-            robotInPolygon(
-                0,
-                Rectangle(world_ptr->field().friendlyNegativeYQuadrant().expand(
-                    Vector(-0.2, 0))),
-                world_ptr, yield);
+            robotInPolygon(0, Rectangle(world_ptr->field().friendlyNegativeYQuadrant()),
+                           world_ptr, yield);
```
Comment (nimazareian) [inline src/software/ai/hl/stp/tactic/goalie/simulated_goalie_tactic_test.cpp:148]: Do you know if there was a reason for expanding the rectangle, previously? Should robot 0 not leave the field?

**Review by nimazareian — COMMENTED** [review]
Overall looks good. Just had two questions

File: src/software/ai/hl/stp/tactic/goalie/simulated_goalie_tactic_test.cpp (hunk 2 above)
Comment (jonathanlew) [reply src/software/ai/hl/stp/tactic/goalie/simulated_goalie_tactic_test.cpp:148]: counterintuitively, I think this expands the rectangle in the -x direction, so this is a tighter constraint and it still passes

File: src/software/geom/polygon.cpp (hunk 1 above)
Comment (jonathanlew) [reply src/software/geom/polygon.cpp:74]: it's confusing to use and not well used (single direction expands was only used in one place)

File: src/software/ai/hl/stp/tactic/goalie/simulated_goalie_tactic_test.cpp (hunk 2 above)
Comment (nimazareian) [reply src/software/ai/hl/stp/tactic/goalie/simulated_goalie_tactic_test.cpp:148]: Interesting

**Review by nimazareian — APPROVED** [review]
LGTM

**Review by akhilveeraghanta — APPROVED** [review]

**Comment by SpeedyPetey** [comment]
lgtm

**Pull request merged**